components/table/hooks/useSorter.tsx

```diff
@@ -127,20 +127,39 @@ function injectSorter<RecordType>(
       const sorterState = sorterStates.find(({ key }) => key === columnKey);
       const sorterOrder = sorterState ? sorterState.sortOrder : null;
       const nextSortOrder = nextSortDirection(sortDirections, sorterOrder);
-      const upNode: React.ReactNode = sortDirections.includes(ASCEND) && (
-        <CaretUpOutlined
-          className={classNames(`${prefixCls}-column-sorter-up`, {
-            active: sorterOrder === ASCEND,
-          })}
-        />
-      );
-      const downNode: React.ReactNode = sortDirections.includes(DESCEND) && (
-        <CaretDownOutlined
-          className={classNames(`${prefixCls}-column-sorter-down`, {
-            active: sorterOrder === DESCEND,
-          })}
-        />
-      );
+
+      let sorter: React.ReactNode;
+      if (column.sortIcon) {
+        sorter = column.sortIcon(sorterOrder);
```

Comment:
```suggestion
        sorter = column.sortIcon({ sorterOrder });
```

Incase that we could add more argument in future.

Reply:
I've fixed it.
https://github.com/ant-design/ant-design/pull/42498/commits/2935cdb0db0a2c72bf2344392a339c2ef3c24a33